controllers/scripts/create_pod_status_patch.py

```diff
@@ -1,195 +1,267 @@
+    if return_value != 0:
+        raise OSError(f"Execution Failed - command: {cmd}")
+
+
+def update_status(pod_name, metadata, volumes):
+    with open("aerospikeConfHash", mode="r") as f:
+        conf_hash = f.read()
+    with open("networkPolicyHash", mode="r") as f:
+        network_policy_hash = f.read()
+    with open("podSpecHash", mode="r") as f:
+        pod_spec_hash = f.read()
+
+    metadata.update({
+        "initializedVolumes": volumes,
+        "aerospikeConfigHash": conf_hash,
+        "networkPolicyHash": network_policy_hash,
+        "podSpecHash": pod_spec_hash,
+    })
+    for pod_addr_name, conf_addr_name in ADDRESS_TYPE_NAME.items():
+        metadata["aerospike"][conf_addr_name] = get_endpoints(address_type=pod_addr_name)
+
+    payload = [{"op": "replace", "path": f"/status/pods/{pod_name}", "value": metadata}]
+    print(40 * "#" + " payload " + 40 * "#")
+    pp(payload)
+    print(89 * "#")
+    with open("/tmp/patch.json", mode="w") as f:
+        json.dump(payload, f)
+        f.flush()
+
+
+def main():
+    try:
+        parser = argparse.ArgumentParser()
+        parser.add_argument("--pod-name", type=str, required=True, dest="pod_name")
+        parser.add_argument("--config", type=str, required=True, dest="config")
+        args = parser.parse_args()
+        config = json.loads(args.config)
+        metadata = get_node_metadata()
+        dd = 'dd if=/dev/zero of={volume_path} bs=1M 2> /tmp/init-stderr || grep -q "No space left on device" ' \
+             '/tmp/init-stderr'
+        find = "find {volume_path} -type f -delete"
+        blkdiskard_init = "blkdiscard {volume_path}"
+        blkdiskard_wipe = "blkdiscard -z {volume_path}"
+        next_major_ver = get_image_version(image=config["spec"]["image"])[0]
+        try:
+            image = config["status"]["image"]
+        except KeyError:
+            image = ""
+        print(f"pod-name: {args.pod_name} - Checking if volumes should be wiped")
+        if image:
+            prev_major_ver = get_image_version(image=image)[0]
+            print(f"pod-name: {args.pod_name} - Checking if volumes should be wiped: "
+                  f"next-major-version: {next_major_ver} prev-major-version: {prev_major_ver}")
+            if (next_major_ver >= BASE_WIPE_VERSION > prev_major_ver) or \
+                (next_major_ver < BASE_WIPE_VERSION <= prev_major_ver):
+                print(f"pod-name: {args.pod_name} - Volumes should be wiped")
+                volumes = process_volumes(pod_name=args.pod_name,
+                                          dd=dd, find=find,
+                                          blkdiskard=blkdiskard_wipe,
+                                          effective_method_key="effectiveWipeMethod",
+                                          volumes=get_volumes(pod_name=args.pod_name, config=config))
+                update_status(pod_name=args.pod_name, metadata=metadata, volumes=volumes)
+                return
```

Comment:
Use if-else here instead of return. The common post-processing section, update_status should be common and written only once. It's possible there will be more follow-up common steps.  It's also hard to quickly see what is going on.